packages/ui-dom-utils/src/findDOMNode.ts

```diff
@@ -49,7 +63,23 @@ function findDOMNode(el?: UIElement) {
   ) {
     return node as Node | Window
   } else if (node) {
-    return ReactDOM.findDOMNode(node as React.ReactInstance) // eslint-disable-line react/no-find-dom-node
+    const reactNode = node as CustomRefNode
+
+    if (isReactNodeWithRef(reactNode)) {
+      const refElement = isRefObject(reactNode.ref)
+        ? reactNode.ref.current
+        : reactNode.ref
+      if (!(refElement instanceof HTMLElement)) {
+        return findDOMNode(refElement)
+      }
+      return refElement
+    } else {
+      console.warn(
+        'Warning: ReactDOM.findDOMNode is deprecated in Strict mode, consider using refs instead.\n See more here: https://instructure.design/#focus-management'
+      )
+
+      return ReactDOM.findDOMNode(node as ReactInstance)!
+    }
   }
```

Comment:
This is the main change:
Our own `findDOMNode` utility was refactored in a way that it is now tries to utilize react refs instead of directly calling `ReactDOM.findDOMNode`.
If a react instance has the `.ref` property on it then this function will try to get the underlying HTML element. 
If the `ref` points to another react instance, then it will recursively try to get the HTML element.

Reply:
I dont understand this part: "See more here: https://instructure.design/#focus-management'"
It does not mention how to use refs